pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -535,14 +535,10 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 	if regional && utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
 		disk, err = gce.getRegionalDiskByName(diskName)
 		if err != nil {
-			glog.V(5).Infof("Could not find regional PD named %q to Attach. Will look for a zonal PD", diskName)
-			err = nil
-		} else {
-			mc = newDiskMetricContextRegional("attach", gce.region)
+			return err
```

Comment:
can a unit test be added?

Reply:
Unfortunately I didn't have time to do the refactor for this release, and I'm planning to leave this unit test for either after code freeze (ideally) or the next release. Sorry this is non-ideal :(